src/main/java/com/conveyal/gtfs/validator/SpeedTripValidator.java

```diff
@@ -37,18 +38,26 @@
     private static final Logger LOG = LoggerFactory.getLogger(SpeedTripValidator.class);
     public static final double MIN_SPEED_KPH = 0.5;
     private boolean allTravelTimesAreRounded = true;
-    private Set<NewGTFSError> travelTimeZeroErrors = new HashSet<>();
+    private final Set<NewGTFSError> travelTimeZeroErrors = new HashSet<>();
 
     public SpeedTripValidator(Feed feed, SQLErrorStorage errorStorage) {
         super(feed, errorStorage);
     }
 
     @Override
-    public void validateTrip(Trip trip, Route route, List<StopTime> stopTimes, List<Stop> stops, List<Location> locations) {
-        if (containsFlexLocations(stopTimes, locations)) {
-            // FLEX TODO: The validator needs to be refactored to work with flex locations. A stop time -> stop_id can
-            // either be a stop_id or a location id.
-            LOG.warn("Trip speed not validated because it contains flex locations!");
+    public void validateTrip(
+        Trip trip,
+        Route route,
+        List<StopTime> stopTimes,
+        List<Stop> stops,
+        List<Location> locations,
+        List<LocationGroup> locationGroups
+    ) {
+        if (hasLocationOrLocationGroupReferences(locations, locationGroups)) {
+            registerError(NewGTFSError.forFeed(
+                NewGTFSErrorType.TRIP_SPEED_NOT_VALIDATED,
+                "Trip speed not validated because it contains at least one location or location group.")
```

Comment:
If validating each entity, the trip id and line number should be included in this error report. Otherwise in the UI you see a potentially long list that looks like



Line # | ID | Bad value
-- | -- | --
N/A | N/A | Trip speed not validated because it contains at least one location or location group.
N/A | N/A | Trip speed not validated because it contains at least one location or location group.
... | ... | ...

Reply:
Line number is not available, but have updated to include the trip entity and trip id.